File: simulator/src/edu/berkeley/sixtus/simul/Node.java
package edu.berkeley.sixtus.simul;

import edu.berkeley.sixtus.simul.Cell.SlotType;

public class Node {
	
	public static final int MAX_TIME_SLOT = 101;
	public static final int MAX_CH_OFFSET = 16;
	
	private int id;
	private int numAllocated;
	private int numCollisions;
	private int numIdenticallyAllocated;
	private Cell[][] slotframe   = new Cell[MAX_TIME_SLOT][MAX_CH_OFFSET];
	
	private int numNeighbors;
	
	public Node(int id){
		this.id=id;
		for (int i=0;i<MAX_TIME_SLOT;i++){
			for (int j=0;j<MAX_CH_OFFSET;j++){
				slotframe[i][j]= new Cell(i, j, -1, Cell.SlotType.OFF);
			}	
		}
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public int getNumAllocated() {
		return numAllocated;
	}

	public void setNumAllocated(int numAllocated) {
		this.numAllocated = numAllocated;
	}

	public int getNumCollisions() {
		return numCollisions;
	}

	public void setNumCollisions(int numCollisions) {
		this.numCollisions = numCollisions;
	}

	public void scheduleLink(int slotNumber, int channelOffset,SlotType type, int node) {
        
 		Cell candidate= new Cell(slotNumber, channelOffset, node, type);
 		//check for collisions
 		Cell ce=slotframe[slotNumber][channelOffset];
 		if (ce.getType()==Cell.SlotType.OFF && ce.getTarget()==-1){
 	    	slotframe[slotNumber][channelOffset]=candidate;
 			this.numAllocated++;
 		}		
	}

	public boolean isLinkAvailable(int slotNumber, int channelOffset,SlotType type, int node) {
		Cell candidate= new Cell(slotNumber, channelOffset, node, type);
 		//check for collisions
 		Cell ce=slotframe[slotNumber][channelOffset];
 		if (ce.getType()==Cell.SlotType.OFF && ce.getTarget()==-1){
 	    	 return true;
 		}else{
 			if (ce.compareTo(candidate)==0){
 				//rescheduling the same cell
 				this.numIdenticallyAllocated++;
 				
 				//System.out.println("Collision occurred by allocating the same cell at slotNumber " + slotNumber + " ch.Offset "+ channelOffset + " at Node " + this.id + " with neighbour " + node + " " +ce);
 				return true;
 			}else{
 				this.numCollisions++;
 			    //System.out.println("Collision occurred at slotNumber " + slotNumber + " ch.Offset "+ channelOffset + " at Node " + this.id + " with neighbour " + node + " " +ce);
 				return false;
 			} 
 		}
 	
	}
	
	public int getNumIdenticallyAllocated() {
		return numIdenticallyAllocated;
	}

	public void setNumIdenticallyAllocated(int numIdenticallyAllocated) {
		this.numIdenticallyAllocated = numIdenticallyAllocated;
	}

	public int getNumNeighbors() {
		return numNeighbors;
	}

	public void setNumNeighbors(int numNeighbors) {
		this.numNeighbors = numNeighbors;
	}

	
	
}

File: python/ScheduleSimulator/simulator/Node.py
'''
Created on 27/06/2013

@author: xvilajosana
'''

import Constants as c
import Cell

class Node(object):
    '''
    classdocs
    '''

    def __init__(self, Nid):
        '''
        Constructor
        '''
        self.id = Nid
        self.numAllocated = 0
        self.numCollisions = 0
        self.numIdenticallyAllocated = 0
        self.numNeighbors = 0
        
        self.slotframe = [[Cell.Cell(x, y, c.CELL_TYPE_OFF,-1) for y in xrange(c.MAX_CH_OFFSET)] for x in xrange(c.MAX_TIME_SLOT)]
        

    def getId(self):
        return self.id;

    def setId(self, Nid):
        self.id = Nid
    
    def getNumAllocated(self):
        return self.numAllocated
    
    
    def setNumAllocated(self, numAllocated):
        self.numAllocated = numAllocated;
    
    
    def getNumCollisions(self):
        return self.numCollisions
    
    def setNumCollisions(self, numCollisions) :
        self.numCollisions = numCollisions;


    def scheduleLink(self, slotNumber, channelOffset, Ctype, node) :
            
        candidate = Cell.Cell(slotNumber, channelOffset, Ctype, node);
        #check for collisions
        ce = self.slotframe[slotNumber][channelOffset];
      
        if (ce.getType() == c.CELL_TYPE_OFF and ce.getTarget() == -1):
            self.slotframe[slotNumber][channelOffset] = candidate;
            self.numAllocated = self.numAllocated + 1

    def isLinkAvailable(self, slotNumber, channelOffset, Ctype, node):
        candidate = Cell.Cell(slotNumber, channelOffset, Ctype, node);
        #check for collisions
        ce = self.slotframe[slotNumber][channelOffset];
        if ce.getType() == c.CELL_TYPE_OFF and ce.getTarget() == -1:
            return True;
        else:
            if (ce.compareTo(candidate) == 0):
                # rescheduling the same cell
                self.numIdenticallyAllocated = self.numIdenticallyAllocated + 1
     
                return True
            else:
                self.numCollisions = self.numCollisions + 1
                return False;

    
    def getNumIdenticallyAllocated(self):
        return self.numIdenticallyAllocated
    
    
    def setNumIdenticallyAllocated(self, numIdenticallyAllocated) :
        self.numIdenticallyAllocated = numIdenticallyAllocated;
    
    
    def getNumNeighbors(self):
        return self.numNeighbors
    
    def setNumNeighbors(self, numNeighbors) :
        self.numNeighbors = numNeighbors;


    
